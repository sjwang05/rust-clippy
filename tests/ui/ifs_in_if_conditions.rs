#![allow(clippy::collapsible_if)]
#![warn(clippy::ifs_in_if_conditions)]

fn main() {
    let cond = true;
    let something_else = true;

    if cond {
        println!();
    } else if something_else {
        println!("check FNs");
    } else {
        println!("check");
    }

    if cond {
        println!();
    }

    if cond {
        println!();
    }

    if cond {
        println!();
    }

    let a = 13;

    if a == 13 {
        println!("check FNs 1");
    } else if a == 14 {
        println!("check FNs 2");
    } else {
        println!("check FNs 3");
    }

    if if a == 13 { 10 } else { 0 } > 5 {
        println!("nested if");
    } else if if a == 12 {
        -10
    } else if if a == 13 { 10 } else { 0 } > 5 {
        0
    } else {
        1
    } < -5
    {
        println!("nested else if");
    }

    if if if if if a == 13 { 12 } else { 13 } == 12 { 11 } else { 13 } == 11 {
        10
    } else {
        13
    } == 10
    {
        9
    } else {
        13
    } == 9
    {
        println!("recursion");
    }
}
