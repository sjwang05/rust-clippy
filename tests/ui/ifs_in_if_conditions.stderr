error: `if` expr in `if` condition
  --> $DIR/ifs_in_if_conditions.rs:38:8
   |
LL |     if if a == 13 { 10 } else { 0 } > 5 {
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider assigning the result of the `if` to a variable and using the variable in the condition instead
   = note: `-D clippy::ifs-in-if-conditions` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::ifs_in_if_conditions)]`

error: `if` expr in `if` condition
  --> $DIR/ifs_in_if_conditions.rs:40:15
   |
LL |       } else if if a == 12 {
   |  _______________^
LL | |         -10
LL | |     } else if if a == 13 { 10 } else { 0 } > 5 {
LL | |         0
LL | |     } else {
LL | |         1
LL | |     } < -5
   | |_____^
   |
   = help: consider assigning the result of the `if` to a variable and using the variable in the condition instead

error: `if` expr in `if` condition
  --> $DIR/ifs_in_if_conditions.rs:42:15
   |
LL |     } else if if a == 13 { 10 } else { 0 } > 5 {
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider assigning the result of the `if` to a variable and using the variable in the condition instead

error: `if` expr in `if` condition
  --> $DIR/ifs_in_if_conditions.rs:51:8
   |
LL |       if if if if if a == 13 { 12 } else { 13 } == 12 { 11 } else { 13 } == 11 {
   |  ________^
LL | |         10
LL | |     } else {
LL | |         13
...  |
LL | |         13
LL | |     } == 9
   | |_____^
   |
   = help: consider assigning the result of the `if` to a variable and using the variable in the condition instead

error: `if` expr in `if` condition
  --> $DIR/ifs_in_if_conditions.rs:51:11
   |
LL |       if if if if if a == 13 { 12 } else { 13 } == 12 { 11 } else { 13 } == 11 {
   |  ___________^
LL | |         10
LL | |     } else {
LL | |         13
LL | |     } == 10
   | |_____^
   |
   = help: consider assigning the result of the `if` to a variable and using the variable in the condition instead

error: `if` expr in `if` condition
  --> $DIR/ifs_in_if_conditions.rs:51:14
   |
LL |     if if if if if a == 13 { 12 } else { 13 } == 12 { 11 } else { 13 } == 11 {
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider assigning the result of the `if` to a variable and using the variable in the condition instead

error: `if` expr in `if` condition
  --> $DIR/ifs_in_if_conditions.rs:51:17
   |
LL |     if if if if if a == 13 { 12 } else { 13 } == 12 { 11 } else { 13 } == 11 {
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider assigning the result of the `if` to a variable and using the variable in the condition instead

error: aborting due to 7 previous errors

